File: compositor/src/main/java/org/westford/compositor/core/OutputFactory.java
package org.westford.compositor.core;


import javax.annotation.Nonnull;
import javax.inject.Inject;
import java.util.Collections;

public class OutputFactory {
    @Nonnull
    private final PrivateOutputFactory privateOutputFactory;

    @Inject
    OutputFactory(@Nonnull final PrivateOutputFactory privateOutputFactory) {
        this.privateOutputFactory = privateOutputFactory;
    }

    public Output create(@Nonnull final RenderOutput renderOutput,
                         @Nonnull final String name,
                         @Nonnull final OutputGeometry outputGeometry,
                         @Nonnull final OutputMode outputMode) {
        final Output output = this.privateOutputFactory.create(renderOutput,
                                                               name);
        output.update(Collections.emptySet(),
                      outputGeometry);
        output.update(Collections.emptySet(),
                      outputMode);
        return output;
    }
}

File: compositor/src/main/java/org/westford/compositor/core/OutputFactory.kt
package org.westford.compositor.core


import javax.inject.Inject
import java.util.Collections

class OutputFactory @Inject
internal constructor(private val privateOutputFactory: PrivateOutputFactory) {

    fun create(renderOutput: RenderOutput,
               name: String,
               outputGeometry: OutputGeometry,
               outputMode: OutputMode): Output {
        val output = this.privateOutputFactory.create(renderOutput,
                name)
        output.update(emptySet<WlOutputResource>(),
                outputGeometry)
        output.update(emptySet<WlOutputResource>(),
                outputMode)
        return output
    }
}
